processor/PlanExecutor.go

```diff
@@ -16,6 +16,15 @@ import (
 )
 
 func merge(source map[string]interface{}, extra map[string]interface{}) map[string]interface{} {
+
+	if source == nil {
+		panic("Source map is nil")
```

Comment:
Should we be `panic`ing or should the method return an `error` as well and let the caller check rather than catch?